src/main/java/com/google/api/codegen/transformer/OutputTransformer.java

```diff
@@ -139,41 +119,78 @@ public void run() {
       List<String> config,
       MethodContext context,
       SampleValueSet valueSet,
-      ScopeTable localVars,
+      OutputContext outputContext,
       CallingForm form) {
     Preconditions.checkArgument(
         !config.isEmpty(),
         "%s:%s: print spec cannot be empty",
         context.getMethodModel().getSimpleName(),
         valueSet.getId());
-    String format = config.get(0);
-    ImmutableList.Builder<OutputView.PrintArgView> argsBuilder = ImmutableList.builder();
-    for (String path : config.subList(1, config.size())) {
+    OutputView.StringFormatView formattedString =
+        stringFormatView(context, outputContext, config, valueSet, form);
+    return OutputView.PrintView.newBuilder().formattedString(formattedString).build();
+  }
+
+  private OutputView.WriteFileView writeFileView(
+      OutputSpec.WriteFileStatement config,
+      MethodContext context,
+      SampleValueSet valueSet,
+      OutputContext outputContext,
+      CallingForm form) {
+    OutputView.StringFormatView fileName =
+        stringFormatView(context, outputContext, config.getFileNameList(), valueSet, form);
+    OutputView.VariableView contents =
+        accessor(
+            new Scanner(config.getContents()), context, valueSet, outputContext.scopeTable(), form);
+    Preconditions.checkArgument(
+        contents.type().isStringType() || contents.type().isBytesType(),
+        "Output to file: expected string or bytes, found %s",
+        contents.type().getTypeName());
+    outputContext.fileOutputTypes().add(contents.type());
+    return OutputView.WriteFileView.newBuilder()
+        .fileName(fileName)
+        .contents(contents)
+        .isFirst(!outputContext.hasMultipleFileOutputs())
```

Comment:
So in line 149, the current write-file type is already added to `OutputContext`, so there should be at least one file output when reaching line 153. If there is just one file output, then this one is the first file output; if there are multiple, then this one is not the first file output.

Reply:
Oh, I see. `hasMultipleFileOutputs()` will return different things because the context changes every time `writeFileView()` is called, once for each file. :+1: